### PR TITLE
Updating MediatR and switching to keyed services

### PR DESCRIPTION
**Pull request by jbogard**
Fixes this issue called out by @davidfowl : https://x.com/davidfowl/status/1725559031484555616?s=20

Pretty neat way of approaching the design, too.

### CODE REVIEW

File: src/EventBus/Extensions/EventBusBuilderExtensions.cs [hunk 1]
```diff
@@ -25,34 +25,7 @@ public static IEventBusBuilder ConfigureJsonOptions(this IEventBusBuilder eventB
...
         // handlers for the event type.
 
-        // MediatR assembly scanning breaks with keyed services see https://github.com/jbogard/MediatR/issues/942.
-        // eventBusBuilder.Services.AddKeyedTransient<IIntegrationEventHandler, TH>(typeof(T));
-
-        // Instead, we'll register the handler as transient and use the HandlerTypes on SubscriptionInfo to get all
-        // handlers for the event type.
-        eventBusBuilder.Services.AddTransient<TH>();
-
-        eventBusBuilder.Services.Configure<EventBusSubscriptionInfo>(o =>
```
Comment (davidfowl) [inline src/EventBus/Extensions/EventBusBuilderExtensions.cs:35]: We still need some of this, but only the list of event types.

This part:

o.EventTypes[typeof(T).Name] = typeof(T);

Comment (jbogard) [reply src/EventBus/Extensions/EventBusBuilderExtensions.cs:35]: Ah right. Just don't need the `HandlerTypes` business anymore.

File: src/EventBusRabbitMQ/RabbitMQEventBus.cs [hunk 2]
```diff
@@ -200,16 +200,8 @@ private async Task ProcessEvent(string eventName, string message)
         // REVIEW: This could be done in parallel
 
         // Get all the handlers using the event type as the key
-        // foreach (var handler in scope.ServiceProvider.GetKeyedServices<IIntegrationEventHandler>(eventType))
-
-        _subscriptionInfo.HandlerTypes.TryGetValue(eventType, out var handlerTypes);
```
Comment (davidfowl) [inline src/EventBusRabbitMQ/RabbitMQEventBus.cs:205]: Need this undeleted as well.

Comment (jbogard) [reply src/EventBusRabbitMQ/RabbitMQEventBus.cs:205]: Does the keyed services not replace the whole `HandlerTypes` business? That you don't need to keep track of all the handler types to resolve, you rely on the container instead.

Comment (davidfowl) [reply src/EventBusRabbitMQ/RabbitMQEventBus.cs:205]: Oh you're right here.

**Review by davidfowl — APPROVED** [review]

**Comment by davidfowl** [comment]
Thanks @jbogard !

**Pull request merged**